internal/integration/search_index_prose_test.go

```diff
@@ -472,4 +494,44 @@ func TestSearchIndexProse(t *testing.T) {
 			})
 			assert.ErrorContains(mt, err, "Attribute mappings missing")
 		})
+
+	mt.Run("case 9: Drivers use server default for unspecified name (`default`) and type (`search`)", func(mt *mtest.T) {
+		view := mt.Coll.SearchIndexes()
+		definition := bson.D{
+			{"mappings", bson.D{
+				{"dynamic", true},
+			}},
+		}
+		opts := options.SearchIndexes()
```

Comment:
[blocking] This would have worked without the fix. We should also test the nil options case: 

```go
indexName, err := view.CreateOne(context.Background(), mongo.SearchIndexModel{
	Definition: definition,
	Options:    nil,
})
```